inc/scaip-shortcode-inserter.php

```diff
@@ -70,12 +70,19 @@ function( $blocks, $block ) {
 
 	$output = '';
 
+	$default_count_blocks = [ 'core/paragraph' ];
+
 	foreach ( $parsed_blocks as $block ) {
 		$is_empty = empty( trim( $block['innerHTML'] ) );
 		if ( $is_empty ) {
 			continue;
 		}
 
+		if ( true !== apply_filters( 'scaip_count_block', in_array( $block['blockName'], $default_count_blocks, true ), $block ) ) {
```

Comment:
This `apply_filters()` doesn't need to run inside `foreach ( $parsed_blocks `. If a website is doing something complex or expensive in their filtering here, that expensive operation will happen once for every block, when it need only happen once.

As an alternative, could you do something like this?

```php
$insertable_blocks = array_flip( apply_filters( 'scaip_list_valid_blocks', [ 'core/paragraph' ] ) );

foreach ( $parsed_blocks as $block ) {
	...
	if ( isset( $insertable_blocks[ $block['blockName'] ) ) {
		$output .= render_block( $block );
		continue;
	}
}
```

Here I use `array_flip` to turn an array of values into an array where the former values are now keys. Checking the keys is faster than searching for array values. https://10up.github.io/Engineering-Best-Practices/php/#build-arrays-that-encourage-lookup-by-key-instead-of-search-by-value